### PR TITLE
Library Unit Test

### PR DESCRIPTION
**Pull request by iankiema**
The following files and folder have been added 
-Spec folder 
-person_spec.rb
-rentals_spec.rb
-student_spec.rb
-teacher_spec.rb

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require_relative '../person'
+
+RSpec.describe Person do
+  it 'creates a new person' do
+    person = Person.new(name: 'John', age: 25)
+    expect(person.name).to eq('John')
+    expect(person.age).to eq(25)
+  end
+end
```
Comment (zeff96) [inline spec/person_spec.rb]: - [ ] Hello! 🌟 According to the project [guideline](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md) 📜, you should ensure that your test suite covers all the methods you've defined within the Person class. This means you need to write tests for each method to make sure they work as expected. 🧪✅
- [ ] To meet this requirement, you should create individual test cases for each method within the Person class, checking various scenarios to ensure their correctness. This comprehensive test coverage will help you verify that your Person class is functioning as intended and that each method behaves correctly in different situations. 🧐🔍
- [ ] In summary, make sure you have a dedicated test for every method in the Person class to validate their functionality and adhere to the project guidelines. 🤝🚀

Screenshot from 2023-10-17 15-55-40

File: spec/teacher_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+require_relative '../specialization'
+
+RSpec.describe Teacher do
+  it 'creates a new teacher with specialization' do
+    specialization = Specialization.new('Math')
+    teacher = Teacher.new(name: 'Mr. Smith', age: 35, specialization: specialization)
+    expect(teacher.name).to eq('Mr. Smith')
+    expect(teacher.age).to eq(35)
+    expect(teacher.specialization).to eq(specialization)
+  end
```
Comment (zeff96) [inline spec/teacher_spec.rb:11]: - [ ] Hello! 🌟 According to the project [guideline](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md) 📜, you should ensure that your test suite covers all the methods you've defined within the Teacher class. This means you need to write tests for each method to make sure they work as expected. 🧪✅
- [ ] To meet this requirement, you should create individual test cases for each method within the Teacher class, checking various scenarios to ensure their correctness. This comprehensive test coverage will help you verify that your Teacher class is functioning as intended and that each method behaves correctly in different situations. 🧐🔍
- [ ] In summary, make sure you have a dedicated test for every method in the Teacher class to validate their functionality and adhere to the project guidelines. 🤝🚀

Screenshot from 2023-10-17 15-55-40

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️ ♻️ ♻️ 
Hi @iankiema and @firepicaso , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: All tests under spec folder
:heavy_check_mark: Correct GitFlow

## Required Changes ♻️
- [ ] Hello! 🌟 According to the project [guideline](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md) 📜, you should ensure that your test suite covers all the methods you've defined within all classes. This means you need to write tests for each method to make sure they work as expected. 🧪✅
- [ ] To meet this requirement, you should create individual test cases for each method within each class, checking various scenarios to ensure their correctness. This comprehensive test coverage will help you verify that all your classes are functioning as intended and that each method behaves correctly in different situations. 🧐🔍
- [ ] In summary, make sure you have a dedicated test for every method in all classes to validate their functionality and adhere to the project guidelines. 🤝🚀

Screenshot from 2023-10-17 15-55-40

- [ ] NOTE: Please note that the changes mentioned here apply to all your classes across the project. 📝 I've reviewed all your classes and found that your test cases do not provide complete coverage for all the methods within each class. 🧐📦

- Therefore, it's important to ensure that your test suites encompass all the methods in every class you've implemented. 🧪✅ This means you should create tests for each method within each class, testing different scenarios to guarantee their accuracy and reliability. This approach ensures that your entire codebase is thoroughly tested and aligns with the project's guidelines. 🚀🔍

- In summary, make sure to adjust your test cases in all classes to cover all their methods, even if there were no inline comments for each class. This will help ensure the robustness and correctness of your code. 🤝🔧

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/book_spec.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require_relative '../book'
+
+RSpec.describe Book do
+  it 'creates a new book' do
+    book = Book.new('Title', 'Author')
+    expect(book.title).to eq('Title')
+    expect(book.author).to eq('Author')
+  end
+end
```
Comment (AbelG101) [inline spec/book_spec.rb]: - [ ] Kindly make sure to add a test for the `add_rental` method which is found inside the `Book` class to ensure it works correctly you can check the `rentals` instance variable of the book after adding a rental.

File: spec/classroom_spec.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+require_relative '../classroom'
+
+RSpec.describe Classroom do
+  it 'creates a new classroom' do
+    classroom = Classroom.new('A101')
+    expect(classroom.label).to eq('A101')
+  end
+end
```
Comment (AbelG101) [inline spec/classroom_spec.rb]: - [ ] Please make sure to add a test for the `add_student` method found inside the `Classroom` class to ensure your tests cover all the methods as per the project requirements.

File: spec/rentals_spec.rb
Comment (AbelG101) [inline spec/rentals_spec.rb:1]: - [ ] [OPTIONAL] Although the previous reviewer forgot to mention this I'd suggest you write unit tests for all classes that you created while implementing the Decorator design pattern as it ensures a better test coverage of your project and it's part of the project requirements as well.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @iankiema & @firepicaso 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- All tests are passing ✔
- Correct use of Git Flow ✔
- Well-structured files ✔
- All linter checks are passing ✔
- Descriptive PR title ✔

 ## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mckent05 — APPROVED** [review]
Hi @iankiema, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Good job adding tests to your project 👍 👍 
## Optional suggestions
**OPTIONAL** - Please note that you did not properly implement the `add_rental` method in the person and book class, I suggest you try to implement them properly 😄 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**